consensus/src/consensus_observer/network_message.rs

```diff
@@ -166,8 +163,139 @@ impl ConsensusObserverDirectSend {
 /// OrderedBlock message contains the ordered blocks and the proof of the ordering
 #[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
 pub struct OrderedBlock {
-    pub blocks: Vec<Arc<PipelinedBlock>>,
-    pub ordered_proof: LedgerInfoWithSignatures,
+    blocks: Vec<Arc<PipelinedBlock>>,
+    ordered_proof: LedgerInfoWithSignatures,
+}
+
+impl OrderedBlock {
+    pub fn new(blocks: Vec<Arc<PipelinedBlock>>, ordered_proof: LedgerInfoWithSignatures) -> Self {
+        Self {
+            blocks,
+            ordered_proof,
+        }
+    }
+
+    /// Returns a reference to the ordered blocks
+    pub fn blocks(&self) -> &Vec<Arc<PipelinedBlock>> {
+        &self.blocks
+    }
+
+    /// Returns a copy of the first ordered block
+    pub fn first_block(&self) -> Option<Arc<PipelinedBlock>> {
+        self.blocks.first().cloned()
+    }
+
+    /// Returns a copy of the last ordered block
+    pub fn last_block(&self) -> Option<Arc<PipelinedBlock>> {
+        self.blocks.last().cloned()
+    }
+
+    /// Returns a reference to the ordered proof
+    pub fn ordered_proof(&self) -> &LedgerInfoWithSignatures {
+        &self.ordered_proof
+    }
+
+    /// Returns a reference to the ordered proof block info
+    pub fn proof_block_info(&self) -> &BlockInfo {
+        self.ordered_proof.commit_info()
+    }
+
+    /// Verifies the ordered blocks and returns an error if the data is invalid.
+    /// Note: this does not check the ordered proof.
+    pub fn verify_ordered_blocks(&self) -> Result<(), Error> {
+        // Verify that we have at least one ordered block
+        if self.blocks.is_empty() {
+            return Err(Error::InvalidMessageError(
+                "Received empty ordered block!".to_string(),
+            ));
```

Comment:
This can be an assertion right? I don't think we make empty blocks in consensus.

Reply:
Aah, someone can still send us a message with an empty block and this will get hit 😄